### PR TITLE
Multi region and custom instance size 

### PR DESCRIPTION
**Pull request by arturrez**
resolves https://github.com/ava-labs/avalanche-cli/issues/1112

example of multiregion usage
`./bin/avalanche node create test2  --use-static-ip=false --authorize-access=true --latest-avalanchego-version=true --gcp --region=us-east1-b,us-west1-b --num-nodes=1,1 --fuji --gcp-project avalabs-experimental --gcp-credentials ~/.config/gcloud/application_default_credentials.json`
#################
`./bin/avalanche node create test2  --use-static-ip=true --authorize-access=true --latest-avalanchego-version=true --aws --aws-profile=ddefault --region=us-east-1,us-west-2 --num-nodes=2,2 --fuji`

Region/NodeNum prompt changes 
```
❯ ./bin/avalanche node create test2  --use-static-ip=false --authorize-access=true --latest-avalanchego-version=true --aws --aws-profile=ddefault  --fuji
✔ us-east-2
How many nodes do you want to set up in us-east-2 AWS Region?.: 2
Current selection: [us-east-2]:2
✔ Yes
✔ us-west-2
How many nodes do you want to set up in us-west-2 AWS Region?.: 4
Current selection: [us-east-2]:2 [us-west-2]:4
Use the arrow keys to navigate: ↓ ↑ → ← 
? Would you like to add additional AWS Region?: 
  ▸ No
    Yes
   ```

### CODE REVIEW

File: cmd/nodecmd/create.go [hunk 1]
```diff
@@ -90,9 +81,11 @@ will apply to all nodes in the cluster`,
 	cmd.Flags().BoolVar(&useStaticIP, "use-static-ip", true, "attach static Public IP on cloud servers")
 	cmd.Flags().BoolVar(&useAWS, "aws", false, "create node/s in AWS cloud")
 	cmd.Flags().BoolVar(&useGCP, "gcp", false, "create node/s in GCP cloud")
-	cmd.Flags().StringVar(&cmdLineRegion, "region", "", "create node/s in given region")
+	cmd.Flags().StringSliceVar(&cmdLineRegion, "region", []string{""}, "create node/s in given region")
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/nodecmd/create.go:84]: We probably just need to keep one of those

**Comment by felipemadero** [comment]
> My thought process was I thought we are going to use a map-like value for region and num of nodes in each e.g. --region us-east-1=2,us-east-2=3,us-west-1=4

I have no strong opinion on this. We probably want to keep the previous options (just --num-nodes and --region for the single region case), and maybe we can map options to both. Anyway, no strong opinion. Just keep the single region option available.

File: cmd/nodecmd/create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:84]: done

File: cmd/nodecmd/create.go [hunk 2]
```diff
@@ -114,6 +107,23 @@ func createNodes(_ *cobra.Command, args []string) error {
 	if !useAWS && awsProfile != constants.AWSDefaultCredential {
 		return fmt.Errorf("could not use AWS profile for non AWS cloud option")
 	}
+	if len(cmdLineRegion) != len(numNodes) {
+		return fmt.Errorf("number of regions and number of nodes must be equal")
+	}
+	// set default instance type
+	switch {
+	case nodeType == "default" && useAWS:
+		nodeType = "c5.2xlarge"
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go]: fixed

File: cmd/nodecmd/create_aws.go [hunk 3]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
+			return nil, nil, nil, nil, err
+		}
+		numNodes = utils.SplitComaSeparatedUInt(numNodesStr)
+	}
+	if len(numNodes) != len(regions) {
+		return nil, nil, nil, nil, fmt.Errorf("number of nodes and regions should be same")
+	}
+	for i, region := range regions {
+		if entry, ok := instanceConf[region]; ok {
+			entry.NumNodes = numNodes[i]
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:185]: why putting number of nodes here? seems unrelated to prefix, etc. Why not putting all together in CloudConfig?

File: cmd/nodecmd/create_gcp.go [hunk 4]
```diff
@@ -100,37 +100,38 @@ func getGCPCloudCredentials() (*compute.Service, string, string, error) {
 	return computeService, gcpProjectName, gcpCredentialsPath, err
 }
 
-func getGCPConfig(zone string) (*compute.Service, string, string, string, string, error) {
-	if zone == "" {
+func getGCPConfig(zones []string) (*compute.Service, []string, string, string, string, error) {
+	if len(zones) == 0 {
```
Comment (felipemadero) [inline cmd/nodecmd/create_gcp.go:104]: same here as for aws. maybe we can have just one function to get regions + num nodes

File: cmd/nodecmd/wiz.go [hunk 5]
```diff
@@ -60,9 +60,9 @@ The node wiz command creates a devnet and deploys, sync and validate a subnet in
 	cmd.Flags().BoolVar(&useStaticIP, "use-static-ip", true, "attach static Public IP on cloud servers")
 	cmd.Flags().BoolVar(&useAWS, "aws", false, "create node/s in AWS cloud")
 	cmd.Flags().BoolVar(&useGCP, "gcp", false, "create node/s in GCP cloud")
-	cmd.Flags().StringVar(&cmdLineRegion, "region", "", "create node/s in given region")
+	cmd.Flags().StringSliceVar(&cmdLineRegion, "region", []string{""}, "create node/s in given region")
```
Comment (felipemadero) [inline cmd/nodecmd/wiz.go:63]: use []string{}

Comment (felipemadero) [reply cmd/nodecmd/wiz.go:63]: add a little more to the flag to indicate it is either a region or a list of regions

File: cmd/nodecmd/wiz.go [hunk 6]
```diff
@@ -60,9 +60,9 @@ The node wiz command creates a devnet and deploys, sync and validate a subnet in
...
 	cmd.Flags().BoolVar(&authorizeAccess, "authorize-access", false, "authorize CLI to create cloud resources")
-	cmd.Flags().IntVar(&numNodes, "num-nodes", 0, "number of nodes to create")
+	cmd.Flags().IntSliceVar(&numNodes, "num-nodes", []int{}, "number of nodes to create")
```
Comment (felipemadero) [inline cmd/nodecmd/wiz.go:65]: add a little more to tell it can be a list, and it is one to one to regions

File: cmd/nodecmd/create.go [hunk 7]
```diff
@@ -90,9 +81,10 @@ will apply to all nodes in the cluster`,
 	cmd.Flags().BoolVar(&useStaticIP, "use-static-ip", true, "attach static Public IP on cloud servers")
 	cmd.Flags().BoolVar(&useAWS, "aws", false, "create node/s in AWS cloud")
 	cmd.Flags().BoolVar(&useGCP, "gcp", false, "create node/s in GCP cloud")
-	cmd.Flags().StringVar(&cmdLineRegion, "region", "", "create node/s in given region")
+	cmd.Flags().StringSliceVar(&cmdLineRegion, "region", []string{""}, "create node/s in given region")
 	cmd.Flags().BoolVar(&authorizeAccess, "authorize-access", false, "authorize CLI to create cloud resources")
-	cmd.Flags().IntVar(&numNodes, "num-nodes", 0, "number of nodes to create")
+	cmd.Flags().IntSliceVar(&numNodes, "num-nodes", []int{}, "number of nodes to create")
```
Comment (felipemadero) [inline cmd/nodecmd/create.go:86]: add more info on this is a list possibly, and 1to1 to regions

File: pkg/terraform/gcp/terraform_gcp.go [hunk 8]
```diff
@@ -21,15 +21,25 @@ import (
 )
 
 // SetCloudCredentials sets GCP account credentials defined in service account JSON file
-func SetCloudCredentials(rootBody *hclwrite.Body, zone, credentialsPath, projectName string) error {
-	// zone's format is us-east1-b, region's format is us-east1
-	region := strings.Join(strings.Split(zone, "-")[:2], "-")
+func SetCloudCredentials(rootBody *hclwrite.Body, zones []string, credentialsPath, projectName string) error {
+	// set default provider
+	region := strings.Join(strings.Split(zones[0], "-")[:2], "-")
```
Comment (felipemadero) [inline pkg/terraform/gcp/terraform_gcp.go:26]: dont we want to do once per zone?

File: pkg/terraform/gcp/terraform_gcp.go [hunk 9]
```diff
@@ -130,79 +148,108 @@ func createCustomTokens(tokenName string) hclwrite.Tokens {
...
-		hcl.TraverseAttr{
-			Name: "gcp-node[*]",
-		},
-		hcl.TraverseAttr{
-			Name: "network_interface.0.access_config.0.nat_ip",
-		},
-	})
+func SetOutput(rootBody *hclwrite.Body, zones []string) {
+	for _, zone := range zones {
+		outputEip := rootBody.AppendNewBlock("output", []string{"instance_ips_" + zone})
```
Comment (felipemadero) [inline pkg/terraform/gcp/terraform_gcp.go:234]: why do you want to do this per zone?

File: pkg/models/cloud.go [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+	InstanceIDs   []string
+	PublicIPs     []string
+	Region        string
+	KeyPair       string
+	SecurityGroup string
+	CertFilePath  string
+	ImageID       string
+}
+
+type InstanceConfig struct {
```
Comment (felipemadero) [inline pkg/models/cloud.go:15]: can we have only one kind of config for all of this?

Comment (felipemadero) [reply pkg/models/cloud.go:15]: I mean, unify CloudConfig and InstanceConfig

File: pkg/models/cloud.go [hunk 11]
```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package models
+
+type CloudConfig struct {
```
Comment (felipemadero) [inline pkg/models/cloud.go:5]: should be called RegionConfig?

File: pkg/models/cloud.go [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+	InstanceIDs   []string
+	PublicIPs     []string
+	Region        string
```
Comment (felipemadero) [inline pkg/models/cloud.go:8]: Is this needed?

File: pkg/models/cloud.go [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+type InstanceConfig struct {
+	Prefix            string
+	CertName          string
+	SecurityGroupName string
+	NumNodes          int
+	InstanceType      string
+}
+
+type CloudConfigMap map[string]CloudConfig
```
Comment (felipemadero) [inline pkg/models/cloud.go:23]: Maybe call this CloudConfig, and RegionConfig the former

File: pkg/models/cloud.go [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+
+type InstanceConfigMap map[string]InstanceConfig
+
+// GetRegions returns a slice of strings representing the regions of the CloudConfigMap.
+func (ccm *CloudConfigMap) GetRegions() []string {
```
Comment (felipemadero) [inline pkg/models/cloud.go:28]: maybe this is not needed, and just use maps.Keys whenever needed

File: pkg/models/cloud.go [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+	regions := []string{}
+	for _, cloudConfig := range *ccm {
+		regions = append(regions, cloudConfig.Region)
+	}
+	return regions
+}
+
+// GetRegions returns a slice of strings containing all the regions associated with the InstanceConfigMap.
+func (icm *InstanceConfigMap) GetRegions() []string {
```
Comment (felipemadero) [inline pkg/models/cloud.go:37]: same

Comment (felipemadero) [reply pkg/models/cloud.go:37]: anyway maybe we can unify the structs

File: pkg/models/cloud.go [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+	regions := []string{}
+	for region := range *icm {
+		regions = append(regions, region)
+	}
+	return regions
+}
+
+// GetInstanceIDs returns a slice of instance IDs based on the specified region.
+func (ccm *CloudConfigMap) GetInstanceIDs(region string) []string {
```
Comment (felipemadero) [inline pkg/models/cloud.go:46]: have two methods, one to get all instances and one to get only per region

File: cmd/nodecmd/create.go [hunk 17]
```diff
@@ -309,34 +314,39 @@ func createNodes(_ *cobra.Command, args []string) error {
 // createClusterNodeConfig creates node config and save it in .avalanche-cli/nodes/{instanceID}
 // also creates cluster config in .avalanche-cli/nodes storing various key pair and security group info for all clusters
 // func createClusterNodeConfig(nodeIDs, publicIPs []string, region, ami, keyPairName, certPath, sg, clusterName string) error {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:316]: thank you. leaked comment

File: cmd/nodecmd/create.go [hunk 18]
```diff
@@ -146,59 +155,55 @@ func createNodes(_ *cobra.Command, args []string) error {
...
-	cloudConfig := CloudConfig{}
+	cloudConfig := models.CloudConfigMap{}
 	publicIPMap := map[string]string{}
 	gcpProjectName := ""
 	gcpCredentialFilepath := ""
-	if cloudService == constants.AWSCloudService {
-		// Get AWS Credential, region and AMI
-		ec2Svc, region, ami, err := getAWSCloudConfig(awsProfile, cmdLineRegion)
+	if cloudService == constants.AWSCloudService { // Get AWS Credential, region and AMI
+		regions, ec2Svc, ami, err := getAWSCloudConfig(awsProfile, cmdLineRegion)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:167]: fixed

File: cmd/nodecmd/create.go [hunk 19]
```diff
@@ -146,59 +155,55 @@ func createNodes(_ *cobra.Command, args []string) error {
...
-			publicIPMap, err = awsAPI.GetInstancePublicIPs(ec2Svc, cloudConfig.InstanceIDs)
-			if err != nil {
-				return err
-			}
-		} else {
-			for i, node := range cloudConfig.InstanceIDs {
-				publicIPMap[node] = cloudConfig.PublicIPs[i]
+		for _, region := range regions {
+			if !useStaticIP {
+				publicIPMap, err = awsAPI.GetInstancePublicIPs(ec2Svc[region], cloudConfig[region].InstanceIDs)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:177]: good catch.

File: cmd/nodecmd/create.go [hunk 20]
```diff
@@ -146,59 +155,55 @@ func createNodes(_ *cobra.Command, args []string) error {
...
-			publicIPMap, err = gcpAPI.GetInstancePublicIPs(gcpClient, projectName, zone, cloudConfig.InstanceIDs)
-			if err != nil {
-				return err
-			}
-		} else {
-			for i, node := range cloudConfig.InstanceIDs {
-				publicIPMap[node] = cloudConfig.PublicIPs[i]
+		for _, zone := range zones {
+			if !useStaticIP {
+				publicIPMap, err = gcpAPI.GetInstancePublicIPs(gcpClient, projectName, zone, cloudConfig[zone].InstanceIDs)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:199]: fixed

File: pkg/models/cloud.go (hunk 11 above)
Comment (arturrez) [reply pkg/models/cloud.go:5]: agree

File: pkg/models/cloud.go (hunk 12 above)
Comment (arturrez) [reply pkg/models/cloud.go:8]: removed

File: pkg/models/cloud.go (hunk 13 above)
Comment (arturrez) [reply pkg/models/cloud.go:23]: ok

File: pkg/models/cloud.go (hunk 14 above)
Comment (arturrez) [reply pkg/models/cloud.go:28]: but replace implementation with maps.Keys

File: pkg/models/cloud.go (hunk 16 above)
Comment (arturrez) [reply pkg/models/cloud.go:46]: agree

File: cmd/nodecmd/create.go [hunk 21]
```diff
@@ -90,9 +81,10 @@ will apply to all nodes in the cluster`,
 	cmd.Flags().BoolVar(&useStaticIP, "use-static-ip", true, "attach static Public IP on cloud servers")
 	cmd.Flags().BoolVar(&useAWS, "aws", false, "create node/s in AWS cloud")
 	cmd.Flags().BoolVar(&useGCP, "gcp", false, "create node/s in GCP cloud")
-	cmd.Flags().StringVar(&cmdLineRegion, "region", "", "create node/s in given region")
+	cmd.Flags().StringSliceVar(&cmdLineRegion, "region", []string{""}, "create node/s in given region")
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:84]: good one. fixed

File: cmd/nodecmd/create.go (hunk 7 above)
Comment (arturrez) [reply cmd/nodecmd/create.go:86]: fixed

File: cmd/nodecmd/create.go [hunk 22]
```diff
@@ -114,6 +106,23 @@ func createNodes(_ *cobra.Command, args []string) error {
 	if !useAWS && awsProfile != constants.AWSDefaultCredential {
 		return fmt.Errorf("could not use AWS profile for non AWS cloud option")
 	}
+	if len(cmdLineRegion) != len(numNodes) {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:109]: fixed

File: pkg/utils/common.go [hunk 23]
```diff
@@ -122,6 +123,28 @@ func ConvertInterfaceToMap(value interface{}) (map[string]interface{}, error) {
 	}
 }
 
+// SplitComaSeparatedString splits and trims a comma-separated string into a slice of strings.
+func SplitComaSeparatedString(s string) []string {
+	cl := strings.Split(s, ",")
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/utils/common.go:128]: perfect

File: pkg/utils/common.go [hunk 24]
```diff
@@ -122,6 +123,28 @@ func ConvertInterfaceToMap(value interface{}) (map[string]interface{}, error) {
...
+	for i := range cl {
+		cl[i] = strings.TrimSpace(cl[i])
+	}
+	return cl
+}
+
+// SplitComaSeparatedInt splits a comma-separated string into a slice of integers.
+func SplitComaSeparatedUInt(s string) []int {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/utils/common.go:136]: fixed

File: pkg/terraform/terraform.go [hunk 25]
```diff
@@ -48,23 +48,27 @@ func CheckIsInstalled() error {
...
-	publicIPs := []string{}
-	ipsOutputWoSpace := strings.TrimSpace(string(ipsOutput))
-	// ip and nodeID outputs are bounded by [ and ,] , we need to remove them
-	trimmedPublicIPs := ipsOutputWoSpace[1 : len(ipsOutputWoSpace)-3]
-	splitPublicIPs := strings.Split(trimmedPublicIPs, ",")
-	for _, publicIP := range splitPublicIPs {
-		publicIPWoSpace := strings.TrimSpace(publicIP)
-		// ip and nodeID both are bounded by double quotation "", we need to remove them before they can be used
-		publicIPs = append(publicIPs, publicIPWoSpace[1:len(publicIPWoSpace)-1])
+func GetPublicIPs(terraformDir string, zones []string) (map[string][]string, error) {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/terraform/terraform.go:51]: done

File: pkg/terraform/aws/terraform_aws.go [hunk 26]
```diff
@@ -177,40 +248,42 @@ func SetupInstances(rootBody *hclwrite.Body, securityGroupName string, useExisti
...
-			Name: "aws_node[*]",
-		},
-		hcl.TraverseAttr{
-			Name: "id",
-		},
-	})
 }
 
 // RunTerraform executes terraform apply function that creates the EC2 instances based on the .tf file provided
 // returns a list of AWS node-IDs and node IPs
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:285]: fixed

File: pkg/terraform/aws/terraform_aws.go [hunk 27]
```diff
@@ -227,36 +300,39 @@ func RunTerraform(terraformDir string, useEIP bool) ([]string, []string, error)
...
-	instanceIDs := []string{}
-	instanceIDsOutputWoSpace := strings.TrimSpace(string(instanceIDsOutput))
-	// eip and nodeID outputs are bounded by [ and ,] , we need to remove them
-	trimmedInstanceIDs := instanceIDsOutputWoSpace[1 : len(instanceIDsOutputWoSpace)-3]
-	splitInstanceIDs := strings.Split(trimmedInstanceIDs, ",")
-	for _, instanceID := range splitInstanceIDs {
-		instanceIDWoSpace := strings.TrimSpace(instanceID)
-		// eip and nodeID both are bounded by double quotation "", we need to remove them before they can be used
-		instanceIDs = append(instanceIDs, instanceIDWoSpace[1:len(instanceIDWoSpace)-1])
+func GetInstanceIDs(terraformDir string, regions []string) (map[string][]string, error) {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/terraform/aws/terraform_aws.go:317]: done

File: cmd/nodecmd/create_aws.go [hunk 28]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
 			[]string{usEast1, usEast2, usWest1, usWest2, customRegion},
 		)
 		if err != nil {
-			return nil, "", "", err
+			return nil, nil, nil, err
 		}
-		if region == customRegion {
-			region, err = app.Prompt.CaptureString("Which AWS region do you want to set up your node in?")
+		if userRegion == customRegion {
+			userRegionList, err := app.Prompt.CaptureString("Which AWS region do you want to set up your node in? Use comma to separate multiple regions")
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:125]: fixed

[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:125]: I am not sure how to make it better for the user input. I think coma separated list is fine as it's consistent with cmdline param

File: cmd/nodecmd/create_aws.go [hunk 29]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
-	}
-	ec2Svc := ec2.New(sess)
-	ami, err := awsAPI.GetUbuntuAMIID(ec2Svc)
-	if err != nil {
-		if strings.Contains(err.Error(), "RequestExpired: Request has expired") {
-			printExpiredCredentialsOutput(awsProfile)
+	ec2SvcMap := map[string]*ec2.EC2{}
+	amiMap := map[string]string{}
+	for _, region := range regions {
+		sess, err := getAWSCloudCredentials(awsProfile, region)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:136]: it's not an issue as we save user input to grant us access as setting.
I suggest putting a ticket and address it in separate PR

File: cmd/nodecmd/create_aws.go [hunk 30]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		ec2SvcMap[region] = ec2.New(sess)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:140]: will be addressed in TF replacement PR. as auth flow and related things will be a little bit different there

File: cmd/nodecmd/create_aws.go (hunk 3 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:185]: agree. fixed

File: cmd/nodecmd/create_aws.go [hunk 31]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
 	return certName, newKeyPairName, nil
 }
 
-func getAWSCloudConfig(awsProfile string, region string) (*ec2.EC2, string, string, error) {
-	if region == "" {
+func getAWSCloudConfig(awsProfile string, regions []string) ([]string, map[string]*ec2.EC2, map[string]string, error) {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:109]: seems reasonable. not going to address it just yet

File: cmd/nodecmd/create_aws.go [hunk 32]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
+	if len(regions) == 0 {
 		var err error
 		usEast1 := "us-east-1"
 		usEast2 := "us-east-2"
 		usWest1 := "us-west-1"
 		usWest2 := "us-west-2"
 		customRegion := "Choose custom region (list of regions available at https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html)"
-		region, err = app.Prompt.CaptureList(
+		userRegion, err := app.Prompt.CaptureList(
 			"Which AWS region do you want to set up your node in?",
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:118]: fixed

File: pkg/terraform/gcp/terraform_gcp.go (hunk 9 above)
Comment (arturrez) [reply pkg/terraform/gcp/terraform_gcp.go:234]: because they have unique names for tf resources

File: pkg/terraform/gcp/terraform_gcp.go (hunk 8 above)
Comment (arturrez) [reply pkg/terraform/gcp/terraform_gcp.go:26]: some resources don't have zone, so instead of discovering what zones are avaliable to change them, I just provide default provider - makes things simpler imho

File: cmd/nodecmd/wiz.go (hunk 6 above)
Comment (arturrez) [reply cmd/nodecmd/wiz.go:65]: fixed.

File: cmd/nodecmd/wiz.go (hunk 5 above)
Comment (arturrez) [reply cmd/nodecmd/wiz.go:63]: fixed

File: cmd/nodecmd/create_aws.go [hunk 33]
```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
...
 	}
-	ux.Logger.PrintToUser("Creating new EC2 instance(s) on AWS...")
-	var useExistingKeyPair bool
-	keyPairExists, err := awsAPI.CheckKeyPairExists(ec2Svc, keyPairName)
-	if err != nil {
-		return nil, nil, "", "", err
+
+	if len(numNodes) == 0 {
+		var err error
+		numNodesStr, err := app.Prompt.CaptureValidatedString("How many nodes do you want to set up on AWS?. Please use comma to separate multiple numbers in case of multiple nodes", func(input string) error {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:168]: moved

File: cmd/nodecmd/create_gcp.go (hunk 4 above)
Comment (arturrez) [reply cmd/nodecmd/create_gcp.go:104]: agree

File: cmd/nodecmd/create_aws.go (hunk 29 above)
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/nodecmd/create_aws.go:129]: Could you make the ticket?

File: cmd/nodecmd/create.go [hunk 34]
```diff
@@ -642,3 +669,68 @@ func requestCloudAuth(cloudName string) error {
...
+		if err != nil {
+			return nil, err
+		}
+		if userRegion == awsCustomRegion {
+			userRegion, err = app.Prompt.CaptureString(fmt.Sprintf("Which %s do you want to set up your node in?", supportedClouds[cloudName].locationName))
+			if err != nil {
+				return nil, err
+			}
+		}
+		numNodes, err := app.Prompt.CaptureUint32(fmt.Sprintf("How many nodes do you want to set up in %s %s?.", userRegion, supportedClouds[cloudName].locationName))
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:722]: fixed

File: cmd/nodecmd/create_aws.go (hunk 29 above)
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:129]: https://github.com/ava-labs/avalanche-cli/issues/1261

File: cmd/nodecmd/create.go [hunk 35]
```diff
@@ -642,3 +668,65 @@
...
+		if err != nil {
+			return nil, err
+		}
+		nodes[userRegion] = int(numNodes)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:729]: fixed

**Review by felipemadero — APPROVED** [review]

File: pkg/aws/aws.go [hunk 36]
```diff
@@ -111,6 +110,7 @@ func GetInstancePublicIPs(ec2Svc *ec2.EC2, nodeIDs []string) (map[string]string,
 		return nil, err
 	}
 	reservations := instanceResults.Reservations
+	fmt.Println(reservations)
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/aws/aws.go:113]: fixed

**Comment by arturrez** [comment]
> Is there custom size work in this PR? doesn't seem like it

@sukantoraymond yes, `cmd.Flags().StringVar(&nodeType, "node-type", "default", "cloud instance type")`
